Dalamud/Game/Text/SeStringHandling/SeString.cs

```diff
@@ -306,7 +332,20 @@ public static SeString CreateMapLink(uint territoryId, uint mapId, float xCoord,
     /// <param name="yCoord">The human-readable y-coordinate for this link.</param>
     /// <param name="fudgeFactor">An optional offset to account for rounding and truncation errors; it is best to leave this untouched in most cases.</param>
     /// <returns>An SeString containing all of the payloads necessary to display a map link in the chat log.</returns>
-    public static SeString? CreateMapLink(string placeName, float xCoord, float yCoord, float fudgeFactor = 0.05f)
+    public static SeString? CreateMapLink(string placeName, float xCoord, float yCoord, float fudgeFactor = 0.05f) =>
+        CreateMapLink(placeName, null, xCoord, yCoord, fudgeFactor);
+    
+    /// <summary>
+    /// Creates an SeString representing an entire Payload chain that can be used to link a map position in the chat log, matching a specified zone name.
+    /// Returns null if no corresponding PlaceName was found.
+    /// </summary>
+    /// <param name="placeName">The name of the location for this link.  This should be exactly the name as seen in a displayed map link in-game for the same zone.</param>
+    /// <param name="instance">An optional area instance number to be included in this link.</param>
+    /// <param name="xCoord">The human-readable x-coordinate for this link.</param>
+    /// <param name="yCoord">The human-readable y-coordinate for this link.</param>
+    /// <param name="fudgeFactor">An optional offset to account for rounding and truncation errors; it is best to leave this untouched in most cases.</param>
+    /// <returns>An SeString containing all of the payloads necessary to display a map link in the chat log.</returns>
+    public static SeString? CreateMapLink(string placeName, int? instance, float xCoord, float yCoord, float fudgeFactor = 0.05f)
```

Comment:
Please rename anything using `instance` to `CreateMapInstanceLink` or `CreateInstancedMapLink`. Keeping the current order and nullability is fine I think, but this is more explicit and likely easier to understand at a glance.

For example, `CreateMapLink("Bozja", currentInstance, mod.PosX, mob.PosY);` will resolve to `placeName, xCoord, yCoord, fudgeFactor` despite very clear variables saying that's... not what it does. IDEs will yell at you for this, but relying on IDEs to catch mistakes like this isn't particularly fun imo.

We'll look at merging them together properly in API X.

Reply:
done and tested! ^w^

```csharp
            this.chatGui.Print(SeString.CreateMapLink(817, 495, 10.6f, 18.2f));
            this.chatGui.Print(SeString.CreateMapLink(817, 495, 43, 136));
            this.chatGui.Print(SeString.CreateMapLink("The Rak'tika Greatwood", 10.6f, 18.2f));
            this.chatGui.Print(SeString.CreateMapLinkWithInstance(817, 495, 6, 10.6f, 18.2f));
            this.chatGui.Print(SeString.CreateMapLinkWithInstance(817, 495, 1, 43, 136));
            this.chatGui.Print(SeString.CreateMapLinkWithInstance("The Rak'tika Greatwood", 3, 10.6f, 18.2f));
```

![image](https://github.com/goatcorp/Dalamud/assets/1133653/111024ec-2765-47c7-be9f-b0aa16ac78c4)